pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -591,7 +599,15 @@ func tryGetLatestObject(informer filter.FilteredSharedIndexInformer, obj interfa
 
 // HasSynced returns true after the initial state synchronization
 func (c *Controller) HasSynced() bool {
-	if !c.initialized.Load() {
+	if c.syncTimeout != nil && c.syncTimeout.Load() {
```

Comment:
`syncTimeout` is set in the `controller.Options` – we only set that when the `secretcontroller` creates the cluster. When we manually trigger `AddMemberCluster` in `initKubeRegistry` this will be unset.

Reply:
I wanted to avoid having multiple places with the logic for timeouts

Reply:
I see, thanks